elytron/WFLY-16306-keycloak-saml-feature-pack.adoc

```diff
@@ -0,0 +1,331 @@
+= WFLY-16306: Keycloak Galleon Feature Pack
+:author:            Farah Juma
+:email:             fjuma@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+It's currently possible to secure applications deployed to WildFly with SAML by installing the
+https://www.keycloak.org/docs/latest/securing_apps/#jboss-eap-wildfly-adapter-2[Keycloak Elytron SAML adapter].
+
+The general installation approach is to unpack the Keycloak adapter zip in the WildFly installation root directory.
+This installs some modules and provides some CLI scripts to complete the configuration. The CLI scripts can then
+be executed to install the `keycloak-saml` subsystem and update the default `application-security-domain` mapping in
+the `undertow` subsystem to use a newly installed `http-authentication-factory` which supports SAML.
+
+The following gist shows the changes that are made to the server by installing the Keycloak Elytron SAML adapter:
+https://gist.github.com/fjuma/97ca51d2d0a380fa6c93664de4d09341
+
+This task is to introduce a new Galleon feature pack that can be used to install the Keycloak Elytron SAML adapter.
+A similar feature pack to install the Keycloak Elytron OIDC adapter was added a little while ago before WildFly
+added native support for OpenID Connect (see https://issues.jboss.org/browse/WFLY-13706[WFLY-13706] / https://issues.jboss.org/browse/KEYCLOAK-14953[KEYCLOAK-14953]).
+Thus, the analysis for this task is quite similar to the analysis for https://issues.jboss.org/browse/WFLY-13706[WFLY-13706].
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-16306[WFLY-16306 - Galleon feature pack to install Keycloak adapters]
+* https://github.com/keycloak/keycloak/issues/12363[Issue #12363 - Galleon feature pack to install Keycloak Elytron SAML adapter]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1928[EAP7-1928]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+
+TBD.
+
+Note that since this feature is being developed across two different projects, we'll need engineers and QE
+from both the WildFly and Keycloak projects involved.
+
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+ * https://github.com/keycloak/keycloak
+ * https://github.com/wildfly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+Note that the code for the Keycloak Elytron SAML adapter is going to be extracted from the https://github.com/keycloak/keycloak repo
+and moved to a separate repo. We don't have details yet on when this will happen. We'll likely need to add our feature pack to the
```

Comment:
@fjuma , @jfdenise can we add these details now?

Reply:
@tommaso-borgato The Keycloak team hasn't extracted the Elytron SAML adapter code from the https://github.com/keycloak/keycloak repo. So our feature pack is currently part of that repo:

https://github.com/keycloak/keycloak/tree/main/distribution/galleon-feature-packs/saml-adapter-galleon-pack